app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,42 @@
+class CommentsController < ApplicationController
+
+  def index
+    @post = Post.find(params[:post_id])
+    @comments = @post.comments.all
+  end
+  def show
+      @post = Post.find(params[:post_id])
+      @comment = Comment.find(params[:id])
+  end
+  def new
+    @post = Post.find(params[:post_id])
+    @comment = Comment.new
+  end
+  def create
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.create(comment_params)
```

Comment:
If the comment does not get created (either by failing validation or db constraint), you can tell the user by rendering the new page with error messages:

```rb
if @post.comments.create(comment_params)
  redirect_to...
else
  render "new"
end
```